### PR TITLE
Skip Comment Feature

### PR DESCRIPTION
**Pull request by chanaMovshowich**
The feature allows for adding special comments above resources that users wish to exclude from tagging, enabling skipping tagging for specific resources, the special comments are: "yor:skip" to skip a specific resource and "yor:skipAll" to skip all resources on the page.
It addresses the difficulties encountered in tagging resources within large and intricate projects, streamlining the process and reducing potential errors.
fixes: https://github.com/bridgecrewio/yor/issues/531

### CODE REVIEW

**Comment by chanaMovshowich** [comment]
@ChanochShayner can you review it please?
thanks

**Comment by ChanochShayner** [comment]
> @ChanochShayner can you review it please? thanks

Thank you for your contribution!
I'll take a look.
Please take a look at the failing jobs (Tests & Lint).
It would be very helpful if you could open an issue for this feature with some information and an explanation.

File: src/common/runner/runner.go [hunk 1]
```diff
@@ -43,6 +43,8 @@ type Runner struct {
 
 const WorkersNumEnvKey = "YOR_WORKER_NUM"
 
+var mutex sync.Mutex
```
Comment (ChanochShayner) [inline src/common/runner/runner.go:46]: Please remove it if it is not in use.

File: src/terraform/structure/terraform_parser.go [hunk 2]
```diff
@@ -25,12 +25,13 @@ import (
 )
 
 var ignoredDirs = []string{".git", ".DS_Store", ".idea", ".terraform"}
+var mutex sync.Mutex
```
Comment (ChanochShayner) [inline src/terraform/structure/terraform_parser.go:28]: Please remove it if it is not in use.

**Review by ChanochShayner — COMMENTED** [review]
Looks good!

**Review by achiar99 — APPROVED** [review]

**Comment by ChanochShayner** [comment]
@chanaMovshowich Please add # checkov:skip=CKV_SECRET_80 to src/terraform/structure/terraform_parser.go in line 363 (like line 361), it recognizes the line as a secret.

File: src/terraform/structure/terraform_parser.go [hunk 3]
```diff
@@ -343,7 +360,7 @@ func (p *TerraformParser) modifyBlockTags(rawBlock *hclwrite.Block, parsedBlock
 				// => we should replace it!
 				rawTagsTokens = newTagsTokens // checkov:skip=CKV_SECRET_6 false positive
 			} else {
-				rawTagsTokens = InsertTokens(rawTagsTokens, newTagsTokens[2:len(newTagsTokens)-2])
+				rawTagsTokens = InsertTokens(rawTagsTokens, newTagsTokens[2:len(newTagsTokens)-2]) // checkov:skip=CKV_SECRET_6 false positive
```
Comment (ChanochShayner) [inline src/terraform/structure/terraform_parser.go:363]:
```suggestion
				rawTagsTokens = InsertTokens(rawTagsTokens, newTagsTokens[2:len(newTagsTokens)-2]) // checkov:skip=CKV_SECRET_80 false positive
```

**Review by ChanochShayner — APPROVED** [review]
Very nice work!

**Pull request merged**